### PR TITLE
missing initialize in J1Spin

### PR DESCRIPTION
**Pull request by shivupa**
## Proposed changes

This was missed in #3257. 

## What type(s) of changes does this code introduce?

- Bugfix

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
AMD workstation 
## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
Test this please

**Review by prckent — DISMISSED** [review]
LGTM, subject to CI.

Thanks Shiv! I am resisting commenting on the rest of this older code, which definitely needs some tidying. Clearly this is an improvement though.

**Comment by shivupa** [comment]
Sure! I am open to cleaning stuff up and would appreciate any specific things that need addressed! Either in this or a separate PR.

**Review by ye-luo — CHANGES_REQUESTED** [review]
In the unit test, please add a call to this fixed constructor and call evaluateLog() to validate the results.

File: src/QMCWaveFunctions/tests/test_J1Spin.cpp [hunk 1]
```diff
@@ -111,5 +111,101 @@ TEST_CASE("J1 spin evaluate derivatives Jastrow", "[wavefunction]")
...
+  active.print(std::cout);
+  int nparam = active.size_of_active();
+  REQUIRE(nparam == 4);
+
+  using ValueType = QMCTraits::ValueType;
+  std::vector<ValueType> expected_dlogpsi(nparam);
+  std::vector<ValueType> expected_dhpsioverpsi(nparam);
+  j1->evaluateDerivatives(elec_, active, expected_dlogpsi, expected_dhpsioverpsi);
+
+  // Check clone constructor
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_J1Spin.cpp:199]: Could you fuse this test into the above one?

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**